### PR TITLE
Split mono_method_get_header_checked into internal and external forms.

### PR DESCRIPTION
**Pull request by jaykrell**
mono_method_get_header_checked is public and must initialize the MonoError.
mono_method_get_header_internal can instead use regular error_init.

### CODE REVIEW

File: mono/metadata/Makefile.am [hunk 1]
```diff
@@ -180,6 +180,7 @@ common_sources = \
 	image-internals.h	\
 	jit-info.c		\
 	loader.c		\
+	loader-internals.h	\
```
Comment (jaykrell) [inline mono/metadata/Makefile.am]: I wonder if we need more levels here, like for functions and opaque types, vs. revealed types and globals. This isn't particularly internal -- it isn't loader internal, just runtime internal.

**Comment by jaykrell** [comment]
The point here is to try to enforce that internal callers always initialize, and not have them just piggyback on top of the initialization that is that there only for externals. A small point I know, but it is the intent now. A few, like 3 or 4, other functions should be split this way.

File: mono/metadata/Makefile.am (hunk 1 above)
Comment (luhenry) [reply mono/metadata/Makefile.am]: We definitely don't need more level 😉

**Comment by jaykrell** [comment]
Zoltan implies I should put this new header content into a preexisting internals files.

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64 Interpreter

**Pull request merged**